pkg/apis/serving/v1beta1/trained_model.go

```diff
@@ -0,0 +1,48 @@
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	duckv1beta1 "knative.dev/pkg/apis/duck/v1beta1"
+)
+
+type TrainedModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              TrainedModelSpec         `json:"spec,omitempty"`
+	Status            TrainedModelDeployStatus `json:"status,omitempty"`
+}
+type TrainedModelSpec struct {
+	// Required field for parent inference service
+	InferenceService string `json:"inferenceService"`
+	// Predictor model spec
+	PredictorModel ModelSpec `json:"predictorModel"`
+	// Explainer model spec
+	ExplainerModel ModelSpec `json:"explainerModel,omitempty"`
+}
+type ModelSpec struct {
+	// Storage URI for the model repository
+	StorageURI string `json:"storageUri"`
+	// Default to latest
+	ModelVersionPolicy ModelVersionPolicy `json:"modelVersionPolicy,omitempty"`
```

Comment:
I'd prefer to leave this out to start, but minimally it should be a pointer (optional)

Reply:
I agree, maybe we should keep the scope a little more contained for the first PR.

Reply:
I think this will be an important field as both tf-serving and triton have version policy. But I am fine leave it out to  restrict the scope.